qiskit_experiments/analysis/curve_fit_analysis.py

```diff
@@ -0,0 +1,244 @@
+            kwargs: additional kwargs to pass to curve_fit.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        return self._run_curve_fit(
+            experiment_data.data,
+            func,
+            p0=p0,
+            p0_func=p0_func,
+            bounds=bounds,
+            fit_mean_data=fit_mean_data,
+            plot=plot,
+            ax=ax,
+            **kwargs,
+        )
+
+    def _run_curve_fit(
+        self,
+        data: List[Dict[str, any]],
+        func: Callable,
+        p0: Optional[List[float]] = None,
+        p0_func: Optional[Callable] = None,
+        bounds: Optional[Tuple] = None,
+        fit_mean_data: bool = False,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+        **kwargs,
+    ):
+        """Run curve fit analysis on circuit data.
+
+        Args:
+            data (ExperimentData): the experiment data to analyze.
+            func: fit function `f(x, *params)`.
+            p0: Optional, initial parameter values for curve_fit.
+            p0_func: Optional, function for calculating initial p0.
+            bounds: Optional, parameter bounds for curve_fit.
+            fit_mean_data: Optional, if True pass means of data points to curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+            kwargs: additional kwargs to pass to curve_fit.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                    ``analysis_results`` may be a single or list of
+                    AnalysisResult objects, and ``figures`` may be
+                    None, a single figure, or a list of figures.
+        """
+        # Compute curve fit data
+        if fit_mean_data:
+            xdata, ydata, sigma, xraw, yraw = average_curve_fit_data(data, return_raw=True)
+        else:
+            xdata, ydata, sigma = curve_fit_data(data)
+
+        if p0_func is not None and p0 is None:
+            p0 = p0_func(xdata, ydata, sigma=sigma)
+
+        # Run curve fit
+        result = curve_fit(func, xdata, ydata, p0, sigma=sigma, bounds=bounds, **kwargs)
+
+        if plot and HAS_MATPLOTLIB:
+            ax = plot_curve_fit(func, result)
+            if fit_mean_data:
+                ax = plot_scatter(xraw, yraw, ax=ax)
+                ax = plot_errorbar(xdata, ydata, sigma, ax=ax)
+            else:
+                ax = plot_scatter(xdata, ydata)
+            figs = [ax]
+        else:
+            figs = None
+
+        return result, figs
+
+
+# NOTE: the following should be handled by DataProcessing module
+
+
+def curve_fit_data(data: List[Dict[str, any]]) -> Tuple[np.ndarray]:
```

Comment:
The data processor in #22 currently does not give back any data. Instead it adds the result of each data processing action to the data dict. I think `curve_fit_data` could be the type of function we need to extract `xvalues` and `yvalues` from processed data.